bundles/org.openhab.core.config.discovery.addon.process/src/main/java/org/openhab/core/config/discovery/addon/process/ProcessAddonFinder.java

```diff
@@ -67,26 +64,27 @@ protected static class ProcessInfo {
          * Initializes the command and commandLine fields.
          * If the command field is not present, it parses the first token in the command line.
          * <p>
-         * Note: the parser could produce a wrong result if BOTH the executable path/filename contains space(s) AND
-         * it contains the identical sub-string as argument[0] after the first space. This is very unlikely, but not
-         * completely impossible. e.g. 'C:\path\to arg0\program.exe arg0 arg1 arg2'
+         * Note: the parser would produce a wrong result if the command line would contain two identical arguments.
+         * e.g. 'C:\path\to file\program.exe arg0 arg0 arg1'. This is very unlikely, but perhaps not impossible.
          */
         protected ProcessInfo(ProcessHandle.Info info) {
             commandLine = info.commandLine().orElse(null);
             String cmd = info.command().orElse(null);
             if ((cmd == null || cmd.isEmpty()) && commandLine != null) {
+                cmd = Objects.requireNonNull(commandLine);
                 String[] args = info.arguments().orElse(null);
                 if (args != null && args.length > 0) {
-                    Matcher whiteSpaceMatcher = WHITE_SPACE.matcher(commandLine);
-                    int iWhite = whiteSpaceMatcher.find() ? whiteSpaceMatcher.start() : 1;
-                    int iArg0 = Objects.requireNonNull(commandLine).indexOf(args[0]);
-                    command = iArg0 > iWhite ? Objects.requireNonNull(commandLine).substring(0, iArg0).trim() : null;
-                } else {
-                    command = Objects.requireNonNull(commandLine).trim();
+                    if (args.length == 1) {
+                        cmd = cmd + " ";
+                    }
+                    int iArg0 = cmd.lastIndexOf(" " + args[0] + " ");
+                    if (iArg0 > 0) {
+                        cmd = cmd.substring(0, iArg0);
+                    }
```

Comment:
Did you try:
```
int iArg0 = cmd.lastIndexOf(String.join(" ", argsArray));
if (iArg0 > 0) {
    cmd = cmd.substring(0, iArg0);
}
```

Reply:
Yes I certainly thought about that already. I was worried that the command line might have extra space characters between the args .. but let me make that change and we can all test it.

Reply:
> Since Windows and Linux are probably the vast majority of user platforms, perhaps the best approach is to use your "String.join(" ", argsArray)" approach, and fix it later if a user on another OS identifies an actual problem. => WDYT?

Yes, agreed.

I was thinking about a solution to overcome multiple whitespace characters between arguments, but that again is only valid if arguments are separated by whitespace characters. If one would find an OS that uses an other character it may not work. But that will probably create many more issues than this.

```
for (int i = argsArray.length - 1; i >= 0; i--) {
    cmd = cmd.substring(0, cmd.lastIndexOf(argsArray[i]);
}
```

Reply:
```
for (int i = argsArray.length - 1; i >= 0; i--) {
    cmd = cmd.substring(0, cmd.lastIndexOf(argsArray[i]);
}
```

^ I think this might also not work in the case that argsArray[m] would be a sub-string of argsArray[n] .. ??

Reply:
Potentially a bit slower indeed, but I doubt the impact will be important.